In [1]:


from scipy import stats
import numpy as np


In [2]:
n=15 # Number of print head taken as sample
P_head= np.array([1.13,1.55,1.43,0.92,1.25,1.36,1.32,0.85,1.07,1.48,1.20,1.33,1.88,1.22,1.29]) # coverting the data into array (In million)

In [3]:
P_head.shape

(15,)

In [4]:

x_bar= P_head.mean() # calculating the sample mean of sample array
s= P_head.std() # calculating the sample standard deviation of sample array

In [5]:
# Using T-distribution  as we don't know the population standard deviation and n <30 
stats.t.interval(0.99,n-1,x_bar,s/np.sqrt(n))

(np.float64(1.097315257758733), np.float64(1.4733514089079334))

In [9]:
t=stats.t.interval(0.99,n-1,x_bar,s/np.sqrt(n))
print(f" Using t-distribution the confidence inverval lies between  {t[0].round(2)} and {t[1].round(2)}" )

 Usint t-distribution the confidence invervar lies between  1.1 and 1.47


In [13]:
# In the b problem the population SD is given as 0.2 million and hence we know the population SD  we can use the Z-distribution  to get the CI 
sigma = 0.2 

stats.norm.interval(0.99,x_bar,sigma/np.sqrt(n))  


(np.float64(1.1523180800624273), np.float64(1.4183485866042391))

In [12]:
z=stats.norm.interval(0.99,x_bar,sigma/np.sqrt(n))  
 
print(f" As we know about the population S.D. hence z-distribution the confidence invervar lies between  {z[0].round(2)} and {z[1].round(2)}" )

 As we know about the population S.D. hence z-distribution the confidence invervar lies between  1.15 and 1.42


##Conclusion#
> Using T- distribution for the given data set the average durability of all the print head ( for 99% confident invterval ) lies between  1.1 and 1.47 (in million of Characters ) 
> 
> For the second problem we know the population standard deviation, hence using z-distribution the average durability of all the print head ( for 99% confident invterval ) lies between  1.15 and 1.42 (in million of Characters )


